src/middlewared/middlewared/plugins/catalogs_linux/items.py

```diff
@@ -200,9 +199,14 @@ def retrieve_item_details(self, item_location, options=None):
                 item_data['healthy_error'] += f'{verror[0]}: {verror[1]}'
 
             # If the item format is not valid - there is no point descending any further into versions
+            if not retrieve_versions:
+                item_data.pop('versions')
             return item_data
 
-        item_data.update(self.item_details(item_location, schema, questions_context))
+        item_data.update(self.item_details(item_location, schema, {
+            'retrieve_latest_version': not retrieve_versions,
```

Comment:
So that just means that we should only retrieve the latest version available and add it to top level item dict instead of traversing all versions. We need to know the latest version of a catalog item, so that means going through at least 1 version per catalog item even when `retrieve_versions` is `False`.

Reply:
Wish `retrieve_versions` was named `retrieve_all_versions` to indicate retrieving all versions for a given catalog item but that key is public facing so I guess not wise to change it now.